spinalcordtoolbox/reports/qc2.py

```diff
@@ -334,15 +337,47 @@ def sct_deepseg_axial(
     ) if fname_seg_lesion else None
     if fname_seg_lesion:
         img_seg_lesion.data = (img_seg_lesion.data > 0.5) * 1
+    img_qc_seg = resample_nib(
+        image=img_qc_seg,
+        image_dest=img_input,
+        interpolation='linear',
+    ) if fname_qc_seg else None
+
+    # Using -qc-seg mask if available, we remove slices which are empty (except for the first 3 and last 3 slices just around the segmented slices)
+    if fname_qc_seg:
+        # We dilate the mask to keep the first and last 3 slices
+        first_slice = min(np.where(img_qc_seg.data)[0]) - 3
+        last_slice = max(np.where(img_qc_seg.data)[0]) + 3
+        # Now we remove the slices from the input image, the spinal cord mask and the lesion mask
+        img_input.data = img_input.data[first_slice:last_slice]
+        img_seg_sc.data = img_seg_sc.data[first_slice:last_slice]
+        if img_seg_lesion:
+            img_seg_lesion.data = img_seg_lesion.data[first_slice:last_slice]
+        img_qc_seg.data = img_qc_seg.data[first_slice:last_slice]
 
     # Each slice is centered on the segmentation
     logger.info('Find the center of each slice')
-    centers = np.array([center_of_mass(slice) for slice in img_seg_sc.data])
+    # Use the -qc-seg mask if available, otherwise use the spinal cord mask
+    if fname_qc_seg:
+        centers = np.array([center_of_mass(slice) for slice in img_qc_seg.data])
+    else:
+        centers = np.array([center_of_mass(slice) for slice in img_seg_sc.data])
```

Comment:
This could be simplified a little bit by using the following pattern:

```suggestion
    # Use the -qc-seg mask if available, otherwise use the spinal cord mask
    img_com = img_qc_seg if fname_qc_seg else img_seg_sc
    centers = np.array([center_of_mass(slice) for slice in img_com.data])
```

Reply:
Nice improvement ! Thanks :)